### PR TITLE
Add dockerfile and Glide files

### PR DESCRIPTION
**Pull request by calavera**
Hi!

I added a couple of changes to the project:

1. A brand new Dockerfile, so people can use go-torch from within
a docker container using `docker run uber/go-torch go-torch ...`

2. I removed the old Godeps file and added Glide files as requested.

### CODE REVIEW

**Comment by sandlerben** [comment]
I've been using Godep for vendoring and it's been working quite well. Do you like bmizerany/vendor better?

**Comment by calavera** [comment]
It's much more simple to use since it leaves all the dependency loading to Go's new vendoring

**Comment by sandlerben** [comment]
Cool!

**Comment by prashantv** [comment]
Thanks for the change @calavera. The docker file seems great.

A couple of things:
1. Can you please fill out the [Uber Contributor License Agreement](https://docs.google.com/a/uber.com/forms/d/1pAwS_-dA1KhPlfxzYLBqK6rsSWwRwH95OCCZrcsY5rk/viewform)
2. Rather than vendor, we prefer to use [glide](https://github.com/Masterminds/glide), and only check in the `glide.yaml` and `glide.lock`.

**Comment by calavera** [comment]
@prashantv I've signed the CLA already and added the Glide files.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM golang:1.6
```
Comment (prashantv) [inline Dockerfile]: do you think it's worth using the alpine image to make this smaller?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+RUN curl -OL https://github.com/Masterminds/glide/releases/download/0.10.1/glide-0.10.1-linux-amd64.tar.gz && \
+    tar -C /usr/bin -xf glide-0.10.1-linux-amd64.tar.gz --strip-components 1 && \
+    git clone --depth=1 https://github.com/brendangregg/FlameGraph.git /opt/flamegraph
+
+COPY . /go/src/github.com/uber/go-torch
+
+RUN cd /go/src/github.com/uber/go-torch && glide install && go install ./...
+
+ENTRYPOINT ["go-torch"]
```
Comment (prashantv) [inline Dockerfile]: assuming /go/bin is already in the PATH?

**Comment by prashantv** [comment]
lgtm, just a couple of small questions

File: Dockerfile (hunk 1 above)
Comment (calavera) [inline Dockerfile]: done, I had to install git and change the tar command, but the resulting image should still be smaller.

File: Dockerfile (hunk 2 above)
Comment (calavera) [inline Dockerfile]: yes, /go/bin is in the path:

```
➜  go-torch git:(dockerfile) docker run uber/go-torch -h
Usage:
  go-torch [OPTIONS]

Application Options:
```

**Comment by prashantv** [comment]
Thanks @calavera!

**Pull request merged**